### PR TITLE
fix(catalog): change route pattern to entities

### PR DESCRIPTION
**Pull request by freben**
## Hey, I just made a Pull Request!

This updates catalog entity routes to match #1947 .

The redirect that was put in place should handle all old `/Component/default:some-name/ci-cd` type routes.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: plugins/catalog/src/components/CatalogPage/CatalogPage.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ import React, { useCallback, useMemo, useState } from 'react';
 import { Link as RouterLink } from 'react-router-dom';
 import { catalogApiRef } from '../../api/types';
 import { EntityFilterGroupsProvider, useFilteredEntities } from '../../filter';
-import { useStarredEntities } from '../../hooks/useStarredEntites';
```
Comment (benjdlambert) [inline plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:34]: ha, nice work

**Review by benjdlambert — APPROVED** [review]

File: plugins/catalog/src/components/CatalogPage/CatalogPage.tsx (hunk 1 above)
Comment (benjdlambert) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:34]: this was totally my spelling of entities.

Comment (freben) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:34]: The vscode spell check extension is great! It's what showed this to me, easy to miss

File: plugins/catalog/src/routes.ts [hunk 2]
```diff
@@ -25,6 +26,17 @@ export const rootRoute = createRouteRef({
...
-  path: ':kind/:optionalNamespaceAndName/*',
+  path: ':namespace/:kind/:name/*',
   title: 'Entity',
 });
+
+// Utility function to get suitable route params for entityRoute, given an
+// entity instance
+export function entityRouteParams(entity: Entity) {
+  return {
+    kind: entity.kind.toLowerCase(),
```
Comment (freben) [inline plugins/catalog/src/routes.ts:37]: i know @stefanalund likes this line (lowercase in routes, all roads led to this one glorious moment)

Comment (stefanalund) [reply plugins/catalog/src/routes.ts:37]: This made my day 😍

**Pull request merged**